extensions/amp-image-slider/0.1/amp-image-slider.js

```diff
@@ -0,0 +1,713 @@
+    this.unlisten_(this.unlistenMouseDown_);
+    this.unlisten_(this.unlistenMouseMove_);
+    this.unlisten_(this.unlistenMouseUp_);
+    this.unlisten_(this.unlistenKeyDown_);
+    this.unregisterTouchGestures_();
+  }
+
+  /**
+   * Get current slider's percentage to the left
+   * Should be wrapped inside measureElement
+   * @private
+   */
+  getCurrentSliderPercentage_() {
+    const {left: barLeft} =
+        this.bar_./*OK*/getBoundingClientRect();
+    const {left: boxLeft, width: boxWidth} = this.getLayoutBox();
+    return (barLeft - boxLeft) / boxWidth;
+  }
+
+  /**
+   * One step left
+   * @param {boolean=} opt_toEnd
+   * @private
+   */
+  stepLeft_(opt_toEnd) {
+    // To the very end of left
+    if (opt_toEnd === true) {
+      this.mutateElement(() => {
+        this.updatePositions_(0);
+      });
+    } else {
+      let newPercentage;
+      this.measureMutateElement(() => {
+        newPercentage = this.limitPercentage_(
+            this.getCurrentSliderPercentage_() - this.stepSize_);
+      }, () => {
+        this.updatePositions_(newPercentage);
+      });
+    }
+  }
+
+  /**
+   * Step to the center
+   * @private
+   */
+  stepExactCenter_() {
+    this.mutateElement(() => {
+      this.updatePositions_(0.5);
+    });
+  }
+
+  /**
+   * One step right
+   * @param {boolean=} opt_toEnd
+   * @private
+   */
+  stepRight_(opt_toEnd) {
+    // To the very end of right
+    if (opt_toEnd === true) {
+      this.mutateElement(() => {
+        this.updatePositions_(1);
+      });
+    } else {
+      let newPercentage;
+      this.measureMutateElement(() => {
+        newPercentage = this.limitPercentage_(
+            this.getCurrentSliderPercentage_() + this.stepSize_);
+      }, () => {
+        this.updatePositions_(newPercentage);
+      });
+    }
+  }
+
+  /**
+   * Move slider based on given pointer x position
+   * Do NOT wrap this in mutateElement!
+   * @param {number} pointerX
+   * @private
+   */
+  pointerMoveX_(pointerX) {
+    const {width, left, right} = this.getLayoutBox();
+    const newPos = Math.max(left, Math.min(pointerX, right));
+    const newPercentage = (newPos - left) / width;
+    this.mutateElement(() => {
+      this.updatePositions_(newPercentage);
+    });
+  }
+
+  /**
+   * Update element positions based on percentage
+   * Should be wrapped inside mutateElement
+   * @param {number} leftPercentage
+   * @private
+   */
+  updatePositions_(leftPercentage) {
+    leftPercentage = this.limitPercentage_(leftPercentage);
+
+    this.updateTranslateX_(this.bar_, leftPercentage);
+    this.updateTranslateX_(this.leftMask_, leftPercentage - 1);
+    this.updateTranslateX_(this.leftAmpImage_, 1 - leftPercentage);
+    if (this.leftLabelWrapper_) {
+      this.updateTranslateX_(this.leftLabelWrapper_, 1 - leftPercentage);
+    }
+  }
+
+  /**
+   * Limit percentage between 0 and 1
+   * @param {number} percentage
+   * @private
+   */
+  limitPercentage_(percentage) {
```

Comment:
What I meant here is that you can use the `clamp` function instead of `limitPercentage_` since it's pretty clear what it is trying to do, but this is fine as is. E.g. 
> `clamp(percentage, /*min*/ 0, /*max*/ 1);`